cpp/ql/src/Critical/DoubleFree.ql

```diff
@@ -16,7 +16,10 @@ import semmle.code.cpp.dataflow.new.DataFlow
 import FlowAfterFree
 import DoubleFree::PathGraph
 
-predicate isFree(DataFlow::Node n, Expr e) { isFree(n, e, _) }
+predicate isFree(DataFlow::Node n, Expr e) {
+  n.asExpr() = e and
+  isFree(_, e, _)
+}
```

Comment:
Hmm, this feels a bit weird now.  It's not really clear that we're ignoring the `n` parameter of `isFree` and finding the node ourselves because we want the pre-update node in this case.

Reply:
That's a good point. In cec785c8cc8ef56c0a0e0e4cfc3fb141f01579cb I've added another column to `isFree/3` now so that we don't have to ignore the `n` parameter now. I think I prefer this. What do you think?

Reply:
I like this solution.